catroid/src/androidTest/java/org/catrobat/catroid/test/common/DeviceVariableAccessorTest.java

```diff
@@ -0,0 +1,180 @@
+	}
+
+	@Test
+	public void saveProjectTest() throws IOException, LoadingProjectException {
+		UserVariable userVariable1 = project.getUserVariable(userVariable.getName());
+		userVariable1.setValue("false");
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		accessor.writeVariable(userVariable1);
+		XstreamSerializer.getInstance().saveProject(project);
+		Project project1 = XstreamSerializer.getInstance().loadProject(project.getDirectory(), context);
+
+		UserVariable result = project1.getUserVariable(userVariable.getName());
+		accessor = new DeviceVariableAccessor(project1.getDirectory());
+		accessor.readUserVariableValue(result);
+
+		assertEquals(result.getDeviceValueKey(), userVariable1.getDeviceValueKey());
+		assertEquals(result.getValue(), userVariable1.getValue());
+	}
+
+	@Test
+	public void copyProjectTest() throws IOException, LoadingProjectException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		accessor.writeVariable(userVariable);
+		XstreamSerializer.getInstance().saveProject(project);
+
+		Project project2 = new Project(context, "projectX2");
+		ProjectCopyTask task = new ProjectCopyTask(null, "projectX2");
+
+		task.task(project.getDirectory(), project2.getName());
+
+		project2 = XstreamSerializer.getInstance().loadProject(project2.getDirectory(), context);
+		project2.getUserVariable(userVariable.getName()).setValue("false");
+
+		UserVariable userVariableProject2 = project2.getUserVariable(userVariable.getName());
+		accessor = new DeviceVariableAccessor(project2.getDirectory());
+		accessor.readUserVariableValue(userVariableProject2);
+
+		cleanupList.add(project2.getName());
+		assertEquals(userVariable.getDeviceValueKey(), userVariableProject2.getDeviceValueKey());
+		assertEquals(userVariable.getValue(), userVariableProject2.getValue());
+	}
+
+	@Test
+	public void deleteLocaleVariableTest() throws IOException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		accessor.writeVariable(localUserVariable);
+		accessor.writeVariable(userVariable);
+		accessor.deleteAllLocalVariables(sprite);
+
+		assertFalse(accessor.readUserVariableValue(localUserVariable));
+		assertTrue(accessor.readUserVariableValue(userVariable));
+	}
+
+	@Test
+	public void cloneSpriteTest() throws IOException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		Sprite clone = new SpriteController().copyForCloneBrick(sprite);
+
+		accessor.writeVariable(localUserVariable);
+
+		UserVariable clonedVar = clone.getUserVariable(localUserVariable.getName());
+		clonedVar.setValue("false");
+		assertTrue(accessor.readUserVariableValue(clonedVar));
+		assertEquals(localUserVariable.getValue(), clonedVar.getValue());
+	}
+
+	@Test
+	public void deleteUserVariableTest() throws IOException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
```

Comment:
with all those tests its kind of hard to tell what portion is setup, and what you actually want to assert.

also, you test accessor with accessor.
please unit test.
-> file doesnt contain UUID -> read returns false
-> file contains UUID -> read returns true
-> file contains UUID -> write updates value
-> file doesnt contain UUID -> write adds new entry with value
-> file doesnt contain UUID -> delete this variable doesnt break anything
-> file contains UUID -> delete really removes it from the file
-> ... etc.

-> if something breaks, you have a unit test pointing out to you what is broken.
currently if either write or read breaks, every test is red, because nothing is tested in separation ("unit").